### PR TITLE
Element.element type by name

### PR DESCRIPTION
**Pull request by SHKnudsen**
### Purpose

Adding Element.ElementTypeByName node to the Elements category and tests

### Reviewers

@radumg 

### FYIs

Not sure if we should add this node to a category of its own.
In the nodes list, this node is called `ElementType.ByName` which indicates it should be under the category `ElementType` but it is grouped under the `Element` category. The `ElementType` category does not exist so we would need to create it.

Will ask about this on the next demo meeting.

#### Dynamo test file:
Element ElementTypeByName

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/Element.cs [hunk 1]
```diff
@@ -761,6 +761,23 @@ private IEnumerable<Element> GetElementSubComponents(Autodesk.Revit.DB.Element e
...
+        /// <param name="name">Name of the type</param>
+        /// <returns>Type Element</returns>
+        public static Element ElementTypeByName(string name)
+        {
+            if (String.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+
+            var elementType = DocumentManager.Instance
+                .ElementsOfType<Autodesk.Revit.DB.ElementType>()
+                .FirstOrDefault(x => x.Name == name).ToDSType(true);
```
Comment (radumg) [inline src/Libraries/RevitNodes/Elements/Element.cs:776]: `FirstOrDefault` returns `null` if it doesn't find anything, so the next chained method `ToDSType` could fail.

If we don't find anything, we should `throw NotFoundException`, which then makes it safe to move `.ToDSType` to the return statement

Comment (radumg) [reply src/Libraries/RevitNodes/Elements/Element.cs:776]: Also, `ToDSType` should normally be on the next line since it's the next method in the chain

File: test/Libraries/RevitNodesTests/Elements/ElementTests.cs [hunk 2]
```diff
@@ -293,5 +293,23 @@ public void CanGetElementSubComponenets()
...
 
         }
+
+        [Test]
+        [TestModel(@".\element.rvt")]
+        public void CanGetElementTypeByName()
+        {
+            // Arrange - get element from model
+            string elementTypeName = "24\" x 24\"";
+            string emptyTypeName = "";
```
Comment (radumg) [inline test/Libraries/RevitNodesTests/Elements/ElementTests.cs:303]: also add a test case for searching for a valid type name that can't be found , which would throw `NotFoundException`

**Review by radumg — CHANGES_REQUESTED** [review]
Looking good, minor changes to catch an edge case

**Comment by SHKnudsen** [comment]
Closing this PR and moving it to a separate one.
